client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -377,6 +378,12 @@ function mediaButton( editor ) {
 				editor.fire( 'SetTextAreaContent', { content } );
 			}
 
+			// We have just replaced some image nodes, potentially losing selection/focus.
+			// So if an image node was selected and one isn't now, re-select it.
```

Comment:
Testing steps are available in https://github.com/Automattic/wp-calypso/issues/14233

cc @msmithgu it's generally nice to copy over testing instructions to the related PR. This lets folks avoid having to context switch to another page and potentially avoid reading a long conversation on what needs to be done about an issue.